docs/benchmarking.md

```diff
@@ -0,0 +1,43 @@
+# Benchmarking Workflows
+
+<!-- used for displaying doctree when selecting file -->
+```{toctree}
+:maxdepth: 4
+```
+
+## Introduction to Benchmarking with Cytosnake
+
+Cytosnake is a Python package known for it  collection of workflows designed for analyzing image-based profiles.
+To ensure these workflows perform at their best, it's essential to conduct benchmarking.
+Benchmarking allows us to assess and optimize the efficiency and reliability of Cytosnake's image analysis processes.
+This tutorial will guide you through the benchmarking process, helping you gauge and improve the performance of these image-based profiling workflows.
+Whether you're a developer seeking to enhance Cytosnake's functionalities or a user interested in its performance, this guide will equip you with the necessary knowledge.
+
+Here in Cytosnake, we use Memray to benchmark our workflows, making it a great tool for tackling memory usage issues, identifying memory leaks, and pinpointing code hotspots that result in excessive allocations.
+Memray's notable features include tracing every function call, handling native calls in C/C++ libraries, minimal application slowdown during profiling, generating diverse reports like flame graphs, compatibility with Python threads, and support for native threads.
+
```

Comment:
Consider adding mentions about why `memray` is used in this way with CytoSnake. For example, what would happen if someone chose to use the command line `memray` without this configuration (would they receive an inaccurate or disorganized result)?

Reply:
added!